backend/soarca/helper_test.go

```diff
@@ -0,0 +1,199 @@
+package soarca
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+)
+
+type TestData struct {
+	Name  string `json:"name"`
+	Value int    `json:"value"`
+}
+
+func TestFetchToJson(t *testing.T) {
+	tests := []struct {
+		name           string
+		setupServer    func() *httptest.Server
+		expectedData   *TestData
+		expectedErrMsg string
+	}{
+		{
+			name: "Successful fetch and unmarshal",
+			setupServer: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Set("Content-Type", "application/json")
+					w.WriteHeader(http.StatusOK)
+					err := json.NewEncoder(w).Encode(TestData{Name: "test", Value: 123})
+					if err != nil {
+						t.Fatalf("could not encode json: %v", err)
+					}
+				}))
+			},
+			expectedData: &TestData{Name: "test", Value: 123},
+		},
+		{
+			name: "Non-200 status code",
+			setupServer: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(http.StatusNotFound)
+				}))
+			},
+			expectedErrMsg: "fetch failed: unexpected status code: 404",
+		},
+		{
+			name: "Invalid JSON",
+			setupServer: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Set("Content-Type", "application/json")
+					w.WriteHeader(http.StatusOK)
+					_, err := w.Write([]byte("Success"))
+					if err != nil {
+						t.Fatalf("Failed to write response: %v", err)
+					}
+				}))
+			},
+			expectedErrMsg: "failed to unmarshal JSON object",
+		},
+		{
+			name: "Empty response body",
+			setupServer: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(http.StatusOK)
+				}))
+			},
+			expectedErrMsg: "fetch failed: empty response body",
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
Logic in tests is something you want to avoid, now you'll need a test to test this logic